### PR TITLE
3P Package scripts for AZSLc

### PR DESCRIPTION
**Pull request by spham-amzn**
- Initial scripts to pull from o3de/o3de-azslc repo and build
- Builds use custom 'test.and.py' scripts which builds by platform and runs tests
- Updated package name from multiplatform **azslc-1.7.23-rev1-multiplatform** to platform-specific  **azslc-1.7.23-rev1-<platform>**
- Minor fixes to pull_and_build_from_git.py

### CODE REVIEW

File: package_build_list_host_darwin.json [hunk 1]
```diff
@@ -27,7 +27,8 @@
         "poly2tri-7f0487a-rev1-mac": "package-system/poly2tri/build_package_image.py --platform-name mac",
         "v-hacd-2.3-1a49edf-rev1-mac": "package-system/v-hacd/build_package_image.py --platform-name mac",
         "SPIRVCross-2021.04.29-rev1-mac": "Scripts/extras/pull_and_build_from_git.py ../../package-system/SPIRVCross --platform-name Mac --package-root ../../package-system --clean",
-        "DirectXShaderCompilerDxc-1.6.2104-o3de-rev2-mac": "Scripts/extras/pull_and_build_from_git.py ../../package-system/DirectXShaderCompiler --platform-name Mac --package-root ../../package-system --clean",
+        "DirectXShaderCompilerDxc-1.6.2104-o3de-rev2-mac": "Scripts/extras/pull_and_build_from_git.py ../../package-system/DirectXShaderCompiler --platform-name Mac --package-root ../../package-system  --clean",
```
Comment (lawsonamzn) [inline package_build_list_host_darwin.json:30]: Eliminate teh extra space before --clean.

Yes, it matters for some reason, makes it not work :)

**Review by lawsonamzn — CHANGES_REQUESTED** [review]

File: package-system/azslc/install_azslc_windows.cmd [hunk 2]
```diff
@@ -0,0 +1,28 @@
+ @echo off
```
Comment (lumberyard-employee-dm) [inline package-system/azslc/install_azslc_windows.cmd:1]: nitpick: extra space before `@echo off`

File: package-system/azslc/build_azslc_linux.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+cd temp/src
+
+# Call the build and test script 
+python3 test.and.py
+
+exit $?
```
Comment (lumberyard-employee-dm) [inline package-system/azslc/build_azslc_linux.sh:13]: The exit call is redundant.
The `python3` invocation return code is the return code of the script

**Review by lumberyard-employee-dm — APPROVED** [review]

File: package_build_list_host_darwin.json (hunk 1 above)
Comment (spham-amzn) [reply package_build_list_host_darwin.json:30]: That was from the previous rev, but i'll fix it anyways :-)

**Review by lawsonamzn — APPROVED** [review]

**Review by galibzon — APPROVED** [review]

**Pull request merged**